### DISCUSSION TITLE
How do you write WP-Cron integration tests?

### DISCUSSION THREAD

**Discussion by jmotes** [dingo-d/wp-pest#32] [Q&A]
I'm having trouble executing scheduled cron events. Is this supported with WP Pest?

For example, I can schedule an event that should run immediately:

```
wp_schedule_single_event(time(), self::ADD_JOBS_HOOK, $args);
```

My tests to ensure the event is scheduled succeeds:
```
$timestamp = wp_next_scheduled(SyncJobsService::ADD_JOBS_HOOK, [['feed_product_id_4'], 'feed_id_4', 'upsert']);
expect($timestamp)->toBeGreaterThan(0);
```

However, when I attempt to run it with `do_action('wp', 'wp-cron')` it doesn't fire off. I've also tried hitting the URL without any success usins this function:

```
function hitWpCronUrl()
{
	$cron_url = site_url('wp-cron.php');
	wp_remote_get($cron_url);
	sleep(5); // Add a delay to ensure cron has time to process
}
```

I'm hoping I'm just doing something wrong and there's a better way to execute the events that I have not yet tried.

**Comment by dingo-d** [comment]
Honestly, I'm not sure you should test the firing of the event (as that will depend on the server anyhow, the wp-cron functionality as is should always work), instead what you should test is the business logic of the cron job itself, and you can do that by just executing the callback function and checking if it does what it needs to do.

At least that's the way I'd write the cron test.
